addon-test-support/@ember/test-helpers/test-metadata.ts

```diff
@@ -0,0 +1,52 @@
+import { BaseContext } from './setup-context';
+
+export interface ITestMetadata {
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+  [key: string]: any;
+
+  readonly isRendering: boolean;
+  readonly isApplication: boolean;
+}
+
+export class TestMetadata implements ITestMetadata {
+  [key: string]: any;
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+
+  constructor() {
+    this.setupTypes = [];
+    this.usedHelpers = [];
+  }
+
+  get isRendering() {
+    return (
+      this.setupTypes.length === 2 &&
```

Comment:
I don't think this is quite right, we shouldn't _require_ that the length is 2 (IOW, why do we care of `this.setupTypes` is `['setupContext', 'setupRenderingContext', 'setupMirage', 'setupEngineContext']`)?
```suggestion
```